### PR TITLE
Issue 3431: Include kubernetes-install.md in documentation under deployment

### LINKED ISSUE #3431: Document update: deployment.md
**Problem description**
1. Content updation
2. Format fix
3. Referencing pravega operator(readme.md)

**Problem location**
documentation/deployment/deployment.md

### LINKED ISSUE #3439: Document inclusion: mkdocs.yml
**Problem description**
 Kubernetes-deployment.md file needs to be include in the mkdocs.yml.

**Problem location**
documentation/mkdocs.yml
documentation/deployment/kubernetes-deployment.md(TBD)

### PR DESCRIPTION
**Pull request by Aparnarr**
**Change log description**  
1. Minor updates were made to deployment.md
2. Kubernetes-install.md file was created and included.
3. File name `kubernetes-install.md` inclusion in mkdocs.yml to render in website.

**Purpose of the change**  
Fixes #3439

### CODE REVIEW

**Comment by Aparnarr** [comment]
@fpj assign @adrianmo for review:

**Comment by Aparnarr** [comment]
@adrianmo requested for review

File: documentation/src/docs/deployment/deployment.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ This guide describes the options for running Pravega for development, testing an
 
 There are two modes for running Pravega.
 
-- Standalone - Standalone mode is suitable for development and testing Pravega applications. It can either be run from the source code, from the distribution package or as a docker container.
+- Standalone - Standalone mode is suitable for development and testing Pravega applications. It can either be run from the source code, from the distribution package or as a docker container. For more information please see the [Running Pravega](http://pravega.io/docs/latest/deployment/run-local/#standalone-mode).
```
Comment (adrianmo) [inline documentation/src/docs/deployment/deployment.md]: Remove "the".

File: documentation/src/docs/deployment/kubernetes-install.md [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+
+- Kubernetes 1.8+
+- An existing Apache Zookeeper 3.5 cluster. This can be easily deployed using our [Zookeeper Operator](https://github.com/pravega/zookeeper-operator).
+- [Pravega Operator](https://github.com/pravega/pravega-operator/edit/master/README.md) manages Pravega clusters deployed to Kubernetes and automates tasks related to operating a Pravega cluster.
+
+## Usage
+
+### Install the Pravega Operator
+
+> Note: If you are running on Google Kubernetes Engine (GKE), please [check this first](#installation-on-google-kubernetes-engine).
```
Comment (adrianmo) [inline documentation/src/docs/deployment/kubernetes-install.md]: Point that link to the GitHub repository.

**Comment by Aparnarr** [comment]
@adrianmo So i can include the content updated via the following PR (https://github.com/pravega/pravega-operator/pull/136).??

**Comment by adrianmo** [comment]
yes, you can also include that.

File: documentation/src/docs/deployment/deployment.md [hunk 3]
```diff
@@ -27,21 +27,22 @@ The following prerequisites are required for running pravega in all modes.
...
-- Bookkeeper 4.4.0
+- Zookeeper 3.5.4-beta
+- Bookkeeper 4.7
 
 For more details on the prerequisites and recommended configuration options for bookkeeper see the [Manual Install Guide](manual-install.md).
 
 ## Installation
 
-There are multiple options provided for running Pravega in different environments. Most of these use the installation package from a Pravega release. You can find the latest Pravega release on the [github releases page](https://github.com/pravega/pravega/releases).
+There are multiple options provided for running Pravega in different environments. Most of these use the installation package from a Pravega release. You can find the latest Pravega release on the [Github Releases](https://github.com/pravega/pravega/releases).
```
[the remark this answers is not in the document]
Comment (Aparnarr) [reply documentation/src/docs/deployment/deployment.md]: @adrianmo Accepted changes

File: documentation/src/docs/deployment/deployment.md (hunk 1 above)
Comment (Aparnarr) [reply documentation/src/docs/deployment/deployment.md]: Accepted suggestion

File: documentation/src/docs/deployment/kubernetes-install.md (hunk 2 above)
Comment (Aparnarr) [reply documentation/src/docs/deployment/kubernetes-install.md]: Edited

**Comment by Aparnarr** [comment]
@fpj  please review and merge

File: documentation/src/docs/deployment/deployment.md [hunk 4]
```diff
@@ -15,33 +15,35 @@ This guide describes the options for running Pravega for development, testing an
 
 There are two modes for running Pravega.
 
-- Standalone - Standalone mode is suitable for development and testing Pravega applications. It can either be run from the source code, from the distribution package or as a docker container.
-- Distributed - Distributed mode runs each component separately on a single or multiple nodes. This is suitable for production in addition for development and testing. The deployment options in this mode include a manual installation, running in a docker swarm or DC/OS.
+- **Standalone:** Standalone mode is suitable for development and testing Pravega applications. It can either be run from the source code, from the distribution package or as a docker container. For more information please see [Running Pravega](http://pravega.io/docs/latest/deployment/run-local/#standalone-mode).
+
+- **Distributed:** Distributed mode runs each component separately on a single or multiple nodes. This is suitable for production in addition for development and testing. The deployment options in this mode include a manual installation, running in a docker swarm or DC/OS.
```
Comment (fpj) [inline documentation/src/docs/deployment/deployment.md]: Aren't we including Kubernetes here? We should add it here, no?

Comment (adrianmo) [reply documentation/src/docs/deployment/deployment.md]: Right, it should be listed there. Maybe replacing DC/OS?

Comment (Aparnarr) [reply documentation/src/docs/deployment/deployment.md]: @fpj, @adrianmo Accepted suggestion

**Comment by Aparnarr** [comment]
@fpj @andreipaduroiu please proceed with merging

File: documentation/src/docs/deployment/deployment.md [hunk 5]
```diff
@@ -15,33 +15,35 @@ This guide describes the options for running Pravega for development, testing an
 
 There are two modes for running Pravega.
 
-- Standalone - Standalone mode is suitable for development and testing Pravega applications. It can either be run from the source code, from the distribution package or as a docker container.
-- Distributed - Distributed mode runs each component separately on a single or multiple nodes. This is suitable for production in addition for development and testing. The deployment options in this mode include a manual installation, running in a docker swarm or DC/OS.
+- **Standalone:** Standalone mode is suitable for development and testing Pravega applications. It can either be run from the source code, from the distribution package or as a docker container. For more information please see [Running Pravega](http://pravega.io/docs/latest/deployment/run-local/#standalone-mode).
+
+- **Distributed:** Distributed mode runs each component separately on a single or multiple nodes. This is suitable for production in addition for development and testing. The deployment options in this mode include a manual installation, running in  Kubernetes or Docker Swarm or DC/OS.
```
Comment (RaulGracia) [inline documentation/src/docs/deployment/deployment.md]: Extra space before Kubernetes.

File: documentation/src/docs/deployment/kubernetes-install.md [hunk 6]
```diff
@@ -0,0 +1,417 @@
+<!--
```
Comment (RaulGracia) [inline documentation/src/docs/deployment/kubernetes-install.md]: If I'm not wrong, this is just the content that can be found in https://github.com/pravega/pravega-operator. The problem of copying it here is that we will need to update this page every time that the operator changes, which is not good for maintainability. I suggest to delete this page and just link the `Kubernetes` entry in table of contents to the appropriate README version of the pravega-operator repository.

Comment (Aparnarr) [reply documentation/src/docs/deployment/kubernetes-install.md]: @RaulGracia  No Raul few sections have been removed and inclusions has been made. So for time being we will link the way as how i have framed.... Sooner once the documentation is all set in the Pravega operator will refer separately from `Pravega operator repo`.

File: documentation/src/docs/deployment/deployment.md (hunk 5 above)
Comment (Aparnarr) [reply documentation/src/docs/deployment/deployment.md]: accepted suggestion

**Comment by Aparnarr** [comment]
@fpj @RaulGracia All the comments adressed

File: documentation/src/docs/deployment/deployment.md [hunk 7]
```diff
@@ -15,33 +15,35 @@ This guide describes the options for running Pravega for development, testing an
...
 
 - Java 8
 
 The following prerequisites are required for running in production. These are only required for running in distributed mode.
 
 - External HDFS 2.7
-- Zookeeper 3.5.1-alpha
-- Bookkeeper 4.4.0
+- Zookeeper 3.5.4-beta
+- Bookkeeper 4.7
```
Comment (RaulGracia) [inline documentation/src/docs/deployment/deployment.md]: Concretely, we are using `4.7.3`.

Comment (Aparnarr) [reply documentation/src/docs/deployment/deployment.md]: Accepted suggestion

File: documentation/src/docs/deployment/kubernetes-install.md [hunk 8]
```diff
@@ -0,0 +1,441 @@
...
+tcp://<cluster-name>-pravega-controller.<namespace>:9090
+```
+
+The `REST` management interface is available at:
+
+```
+http://<cluster-name>-pravega-controller.<namespace>:10080/
+```
+
+[Check this](#direct-access-to-the-cluster) to enable direct access to the cluster for development purposes.
```
Comment (Aparnarr) [inline documentation/src/docs/deployment/kubernetes-install.md]: @adrianmo this reference is not mentioned "#direct-access-to-the-cluster)" i feel it has been updated as 'enable external access':
(https://github.com/pravega/pravega-operator/blob/489e407137769e7e1c63a198a3e9ffebf4adcf45/README.md#enable-external-access)
should i update or remove this....??

Comment (adrianmo) [reply documentation/src/docs/deployment/kubernetes-install.md]: @Aparnarr I updated the README on friday after mergint the upgrade feature. Please do the following:

- Remove the "Direct access to the cluster" section and links
- Add the "enable external access" section
- Add the "Upgrade a Pravega Cluster" section
- Add the "Scale a Pravega Cluster"
- Check that the rest of sections are up to date

Thanks.

Comment (Aparnarr) [reply documentation/src/docs/deployment/kubernetes-install.md]: Doc up-to-date please check

**Comment by Aparnarr** [comment]
@fpj @RaulGracia Please approve this PR and proceed to merging

**Pull request merged**